### DISCUSSION TITLE
Tasmota Thermostat on Shelly UNI device

### DISCUSSION THREAD

**Discussion by dexterbot80** [arendst/Tasmota#19843] [Support request]
Hello,
I compiled last version of Tasmota with thermostat enabled and i use [this method](https://github.com/arendst/mgos-to-tasmota) to convert Shelly UNI to Tasmota.
https://github.com/arendst/mgos-to-tasmota
Now when i tried to update the firmware with my version i received the message: Upload Failed - Not enough space
I tried to upload tasmota_minimal and tasmota_lite but i received the same message. 

I just want to make an simple MQTT thermostat using Shelly UNI device and DS18B20 sensor.
Can someone help me with this?
Thank you
<img width="417" alt="Screenshot 2023-10-26 200929" src="https://github.com/arendst/Tasmota/assets/47325794/0c2905d8-9d74-4cc4-a248-3cdda5267f31">

**Comment by sfromis** [comment]
You did not tell what binary file you tried to flash. When upgrading, you should **always** use the `.gz` variant of the binary. And it can depend on what binary you already had flashed.

**Reply by dexterbot80** [reply]
As usual, @sfromis gives the answers that fit :) Thank you very much proffesor, I succeeded with .gz
